plugins/filesystem/dropbox/Adapter/JoomlaDropboxAdapter.php

```diff
@@ -0,0 +1,388 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: kasun
+ * Date: 6/27/17
+ * Time: 11:16 PM
+ */
+
+namespace Joomla\Plugin\Filesystem\Dropbox\Adapter;
+
+defined('_JEXEC') or die;
+
+\JLoader::import('filesystem.dropbox.vendor.autoload', JPATH_PLUGINS);
+
+use Joomla\CMS\Uri\Uri;
+use Joomla\Component\Media\Administrator\Adapter\AdapterInterface;
+use Joomla\Component\Media\Administrator\Adapter\FileNotFoundException;
+
+/**
+ * Class JoomlaDropboxAdapter
+ * @package Joomla\Plugin\Filesystem\Dropbox\Adapter
+ */
+class JoomlaDropboxAdapter implements AdapterInterface
+{
+	/**
+	 * Supported extension for thumbnails
+	 *
+	 * @var array
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private $supportedThumbnailImageFormats = array('jpg', 'jpeg', 'png', 'tiff', 'tif', 'gif' , 'bmp');
+
+	/**
+	 * Dropbox client to work with
+	 *
+	 * @var \Srmklive\Dropbox\Client\DropboxClient
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private $client = null;
+
+	/**
+	 * Dropbox adapter to work with
+	 *
+	 * @var \Srmklive\Dropbox\Adapter\DropboxAdapter
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private $adapter = null;
+
+	/**
+	 * Flysystem driver
+	 *
+	 * @var \League\Flysystem\Filesystem
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private $dropbox = null;
+
+	/**
+	 * DropboxAdapter constructor.
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct($apiToken)
+	{
+		$this->client  = new \Srmklive\Dropbox\Client\DropboxClient($apiToken);
```

Comment:
Move them to their own function like getClient and call it only when needed. Plugins can be instantiated without even being used.